jetbrains-core/src/software/aws/toolkits/jetbrains/services/sqs/EditAttributesDialog.kt

```diff
@@ -0,0 +1,111 @@
+
+import com.intellij.openapi.application.ModalityState
+import com.intellij.openapi.options.ConfigurationException
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.openapi.ui.ValidationInfo
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import software.amazon.awssdk.services.sqs.SqsClient
+import software.amazon.awssdk.services.sqs.model.QueueAttributeName
+import software.amazon.awssdk.services.sqs.model.SqsException
+import software.aws.toolkits.core.utils.error
+import software.aws.toolkits.core.utils.getLogger
+import software.aws.toolkits.jetbrains.utils.ApplicationThreadPoolScope
+import software.aws.toolkits.jetbrains.utils.getCoroutineUiContext
+import software.aws.toolkits.jetbrains.utils.notifyInfo
+import software.aws.toolkits.resources.message
+import software.aws.toolkits.telemetry.Result
+import software.aws.toolkits.telemetry.SqsTelemetry
+import javax.swing.JComponent
+
+class EditAttributesDialog(
+    private val project: Project,
+    private val client: SqsClient,
+    private val queue: Queue,
+    private val attributes: Map<QueueAttributeName, String>
+) : DialogWrapper(project), CoroutineScope by ApplicationThreadPoolScope("EditAttributesDialog") {
+    val view = EditAttributesPanel()
+
+    init {
+        title = message("sqs.edit.attributes")
+        setOKButtonText(message("sqs.edit.attributes.save"))
+        populateFields()
+        init()
+    }
+
+    override fun createCenterPanel(): JComponent? = view.component
+
+    override fun doValidate(): ValidationInfo? {
+        try {
+            view.messageSize.validateContent()
+            view.retentionPeriod.validateContent()
+        } catch (e: ConfigurationException) {
+            return ValidationInfo(e.title)
+        }
+        return view.waitTime.validate() ?: view.deliveryDelay.validate() ?: view.waitTime.validate()
+    }
+
+    override fun doCancelAction() {
+        SqsTelemetry.editQueueAttributes(project, Result.Cancelled, queue.telemetryType())
+        super.doCancelAction()
+    }
+
+    override fun doOKAction() {
+        if (!isOKActionEnabled) {
+            return
+        }
+        isOKActionEnabled = false
+        launch {
+            try {
+                updateAttributes()
+                notifyInfo(
+                    project = project,
+                    title = message("sqs.service_name"),
+                    content = message("sqs.edit.attributes.updated", queue.queueName)
+                )
+                SqsTelemetry.editQueueAttributes(project, Result.Succeeded, queue.telemetryType())
+                withContext(getCoroutineUiContext(ModalityState.any())) {
+                    close(OK_EXIT_CODE)
+                }
+            } catch (e: SqsException) {
+                LOG.error(e) { "Updating queue attributes failed" }
+                setErrorText(e.message)
```

Comment:
do we need to do this back on the UI context?

Reply:
Nope this updates on the UI context later when called from a background thread